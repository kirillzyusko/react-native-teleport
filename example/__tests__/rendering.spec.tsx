import { render, screen } from "@testing-library/react-native";

import { View } from "react-native";

import { PortalProvider, PortalHost, Portal } from "react-native-teleport";

function TestComponent() {
  return (
    <PortalProvider>
      <PortalHost name="app" />
      <View>
        <Portal hostName="app" name="modal">
          <View />
        </Portal>
      </View>
    </PortalProvider>
  );
}

describe("portal rendering", () => {
  it("should render without errors", () => {
    render(<TestComponent />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
