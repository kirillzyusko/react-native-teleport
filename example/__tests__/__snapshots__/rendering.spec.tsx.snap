// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`portal rendering should render without errors 1`] = `
<PortalProvider>
  <PortalHost
    name="app"
  />
  <View>
    <Portal
      hostName="app"
      name="modal"
    >
      <View />
    </Portal>
  </View>
</PortalProvider>
`;
